src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                book.setId(id);
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create book  " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (PreparedStatement preparedStatement = UtilConnection.getConnection()
+                .prepareStatement(FIND_BY_ID_QUERY)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(createBook(resultSet));
+            } else {
+                return Optional.empty();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (PreparedStatement preparedStatement = UtilConnection.getConnection()
+                .prepareStatement(FIND_ALL_QUERY)) {
+            List<Book> books = new ArrayList<>();
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                books.add(createBook(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find all", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (PreparedStatement preparedStatement = UtilConnection.getConnection()
+                .prepareStatement(UPDATE_QUERY)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+
+            int insertedRows = preparedStatement.executeUpdate();
+            checkInsertedRows(insertedRows);
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book: " + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (PreparedStatement preparedStatement = UtilConnection.getConnection()
+                .prepareStatement(DELETE_QUERY)) {
+            preparedStatement.setLong(1, id);
+
+            int deletedRows = preparedStatement.executeUpdate();
+            return deletedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book by id: " + id, e);
+        }
+    }
+
+    private void checkInsertedRows(int insertedRows) {
+        if (insertedRows < 1) {
+            throw new RuntimeException("Expected at least one row to insert");
+        }
+    }
+
+    private Book createBook(ResultSet resultSet) throws SQLException {
```

Comment:
this method can easily be mistaken with the `create(Book)`, let's name it like so:
```suggestion
    private Book parseBook(ResultSet resultSet) throws SQLException {
```